Bugfix `restic find`: missing check for mtime --oldest/--newest

wplapper opened the pull request:
Add new backup-data.tar.gz which includes in directory 0/0/7 51 files with a mtime date range from 2014 to 2025.

Add missing check for `--oldest` >= `--newest`.

Add test which checks for `--oldest` >= `--newest`.

<!--
Thank you very much for contributing code or documentation to restic! Please
fill out the following questions to make it easier for us to review your
changes.
-->

What does this PR change? What problem does it solve?
-----------------------------------------------------
This PR fixes and issue described in #5280 and fixes the missing test for `--oldest` >= `--newest`.
<!--
Describe the changes and their purpose here, as detailed as needed.
-->

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
See #5280
<!--
Link issues and relevant forum posts here.

If this PR resolves an issue on GitHub, use "Closes #1234" so that the issue
is closed automatically when this PR is merged.
-->

Checklist
---------

<!--
You do not need to check all the boxes below all at once. Feel free to take
your time and add more commits. If you're done and ready for review, please
check the last box. Enable a checkbox by replacing [ ] with [x].

Please always follow these steps:
- Read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- Enable [maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- Run `gofmt` on the code in all commits.
- Format all commit messages in the same style as [the other commits in the repository](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
-->

- [X] I have added tests for all code changes.
~~- [ ] I have added documentation for relevant changes (in the manual).~~
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [X] I'm done! This pull request is ready for review.

wplapper committed:
find: fix missing check for mtime oldest/newest

Add new backup-data.tar.gz which includes in directory 0/0/7
51 files with a mtime date range from 2014 to 2025.

Add missing check for `--oldest` >= `--newest`.

Add test which checks for `--oldest` >= `--newest`.

wplapper commented:
I have decided to revert this PR to closed status to ease the pressure on the 
maintainers of restic. Once the number of PRs has dropped considerably, I might
moved this PR back to the queue to be reviewed. In case you consider this PR
important, please contact me via the usual channels. Thank you!

MichaelEischer commented:
> I have decided to revert this PR to closed status to ease the pressure on the
> maintainers of restic. Once the number of PRs has dropped considerably, I might
> moved this PR back to the queue to be reviewed. In case you consider this PR
> important, please contact me via the usual channels. Thank you!

I'd actually prefer to keep the PRs open to give me the chance to decided which PRs appear more important to me.

wplapper commented:
>   I'd actually prefer to keep the PRs open to give me the chance to decided which PRs appear more important to me.

Michael, it would be good to know your priorities. Then I could work along these important issues. Thanks!

wplapper committed:
restic find - check options for validity

cmd/restic/testdata/backup-data.tar.gz: converted to standard test repository

cmd/restic/cmd_find_integration_test.go:
added missing parameter for 'term'

wplapper committed:
restic find time ordering options - add issue file

MichaelEischer committed:
reword error message and simplify test

MichaelEischer reviewed (approved):
LGTM. Thanks!

I've reworded the error message and the changelog a bit. The test is now also a lot closer. The error is also thrown for an empty repository, so there's no need to create an actual snapshot.
